library/src/main/java/com/mux/player/internal/cache/CacheController.kt

```diff
@@ -149,23 +142,31 @@ internal object CacheController {
   @TargetApi(Build.VERSION_CODES.N)
   private fun closeDatastoreApiN() {
     val totalPlayersNow = playersWithCache.updateAndGet { if (it > 0) it - 1 else it }
-    Log.d(TAG, "closeDatastoreApiN: now have $totalPlayersNow players")
     if (totalPlayersNow == 0) {
       ioScope.launch { datastore.close() }
     }
   }
 
   private fun closeDatastoreLegacy() {
     val totalPlayersNow = playersWithCache.decrementAndGet()
-    Log.d(TAG, "closeDatastoreLegacy: now have $totalPlayersNow players")
     if (totalPlayersNow == 0) {
       ioScope.launch { datastore.close() }
     }
   }
 
+  /**
+   * Returns true if the response must be revalidated before use.
+   *
+   * Requiring revalidation doesn't necessarily mean that the entry needs to be deleted
+   */
+  @JvmSynthetic
+  internal fun revalidateRequired(nowUtc: Long, fileRecord: FileRecord): Boolean {
+    // assume 'immutable' if we didn't get 'no-cache' because it saves a round-trip.
```

Comment:
thought(non-blocking): this is definitely fine for now, based on actual usage and expectations around media segments. A "fancier" version of this would keep track of "refresh" conditions (either app reload or multivariant playlist reload), per https://www.rfc-editor.org/rfc/rfc8246.html, which maybe we want to take note of somewhere but definitely shouldn't concern ourselves with at this point.